ob1k-concurrent/src/main/java/com/outbrain/ob1k/concurrent/Try.java

```diff
@@ -1,44 +1,214 @@
 package com.outbrain.ob1k.concurrent;
 
-import com.google.common.base.Function;
+import java.util.NoSuchElementException;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.function.Predicate;
 
 /**
- * represent either a result or an error using two ADTs - Success and Failure.
+ * Represents a computation that may either result in an exception, or return a successfully computed value.
+ * Similar to Scala's Try.
  *
- * @author aronen
+ * @param <T> the type returned by the computation.
+ * @author marenzon, aronen
  */
 public abstract class Try<T> {
```

Comment:
I actually want this Try to be an interface, but can't right now because of Scala - https://github.com/sbt/sbt/issues/1128
maybe moving Scala code outside again?